provider/cluster/kube/client.go

```diff
@@ -401,99 +402,158 @@ func (c *client) ServiceStatus(ctx context.Context, lid mtypes.LeaseID, name str
 
 		// Get the amount of available CPU, then subtract that in use
-		cpu := knode.Status.Allocatable.Cpu().MilliValue()
-		cpu -= mnode.Usage.Cpu().MilliValue()
-		if cpu < 0 {
-			cpu = 0
+		var tmp resource.Quantity
+
+		tmp = knode.cpu.available()
+		cpuAvailable := (&tmp).MilliValue()
+		if cpuAvailable < 0 {
+			cpuAvailable = 0
 		}
 
-		// Get the amount of memory, then subtract that in use
-		memory := knode.Status.Allocatable.Memory().Value()
-		memory -= mnode.Usage.Memory().Value()
-		if memory < 0 {
-			memory = 0
+		tmp = knode.memory.available()
+		memoryAvailable := (&tmp).Value()
+		if memoryAvailable < 0 {
+			memoryAvailable = 0
 		}
 
-		// Get the amount of storage, then subtract that in use
-		storage := knode.Status.Allocatable.StorageEphemeral().Value()
-		storage -= mnode.Usage.StorageEphemeral().Value()
-		if storage < 0 {
-			storage = 0
+		tmp = knode.storage.available()
+		storageAvailable := (&tmp).Value()
+		if storageAvailable < 0 {
+			storageAvailable = 0
 		}
 
 		resources := types.ResourceUnits{
 			CPU: &types.CPU{
-				Units: types.NewResourceValue(uint64(cpu)),
+				Units: types.NewResourceValue(uint64(cpuAvailable)),
 				Attributes: []types.Attribute{
 					{
 						Key:   "arch",
-						Value: knode.Status.NodeInfo.Architecture,
-						// Value: types.NewAttributeValue(
-						// 	[]string{
-						// 		knode.Status.NodeInfo.Architecture,
-						// 	}),
+						Value: knode.arch,
 					},
 					// todo (#788) other node attributes ?
 				},
 			},
 			Memory: &types.Memory{
-				Quantity: types.NewResourceValue(uint64(memory)),
+				Quantity: types.NewResourceValue(uint64(memoryAvailable)),
 				// todo (#788) memory attributes ?
 			},
 			Storage: &types.Storage{
-				Quantity: types.NewResourceValue(uint64(storage)),
+				Quantity: types.NewResourceValue(uint64(storageAvailable)),
 				// todo (#788) storage attributes like class and iops?
 			},
 		}
 
-		nodes = append(nodes, cluster.NewNode(knode.Name, resources))
+		nodes = append(nodes, cluster.NewNode(nodeName, resources))
 	}
 
 	return nodes, nil
 }
 
-func (c *client) activeNodes(ctx context.Context) (map[string]*corev1.Node, error) {
+type resourcePair struct {
+	allocatable resource.Quantity
+	allocated   resource.Quantity
+}
+
+func (rp resourcePair) available() resource.Quantity {
+	result := rp.allocatable.DeepCopy()
+	// Modifies the value in place
+	(&result).Sub(rp.allocated)
+	return result
+}
+
+type nodeResources struct {
+	cpu     resourcePair
+	memory  resourcePair
+	storage resourcePair
+	arch    string
+}
+
+func (c *client) activeNodes(ctx context.Context) (map[string]nodeResources, error) {
 	knodes, err := c.kc.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return nil, err
 	}
 
-	retnodes := make(map[string]*corev1.Node)
+	listOptions := metav1.ListOptions{
+		FieldSelector: "status.phase!=Failed,status.phase!=Succeeded",
+	}
+	pods, err := c.kc.CoreV1().Pods("").List(ctx, listOptions)
```

Comment:
Do we need to paginate here?

Why is it better to do this than to use `metrics-server`?

Reply:
The metrics-server only returns actual CPU usage on a node, rather than the amount allocated to each pod in the node. For CPU this number is usually zero, which is why the provider would continue to bid on things it couldn't start up.

Reply:
I see, that makes sense.  Thanks.

I agree with an earlier sentiment that you wrote - it probably shouldn't do this so frequently.  Maybe it can be _yet another flag_ or just an environment variable (`caarlos0/env` or similar).

I think that if we do things this way that we might be able to simplify things a bit.  Previously we would:

1. use `metrics-server` to collect current unused capacity.
2. initially reserve capacity in _pending_; available would be `available = unused_capacity - pending`.
3. once a workload was "healthy" aka _active_, it was assumed that it was included in _used_ capacity from the metrics server; hence it would always be that `available = unused_capacity - pending`, not `available = unused_capacity - pending - active`.

All of this assumed that `Resources.Requests` was ignored, and as you found, that is not the case.

Nothing actionable here, just thinking out loud that maybe some complexity can be removed.

Reply:
Pagination has been added here